examples/video.py

```diff
@@ -88,12 +98,46 @@ def load_img(file):
                 else:
                     win.set_windowed()
                     full = 0
+            elif event.key == pygame.K_s:
+                readsurf = renderer.get_surface()
+                pygame.image.save(readsurf, "test.png")
+
             elif event.key == pygame.K_SPACE:
                 bg_index = (bg_index + 1) % len(backgrounds)
                 renderer.draw_color = backgrounds[bg_index]
 
+
+
+    # TODO: use this from_surface somehow.
+    # surf = pg.Surface((64,64))
+    # surf.fill((0,255,0))
+    # # This should draw a green rect
+    # tex = Texture.from_surface(renderer, surf)
+    # tex.draw(None, pg.Rect(64, 64, 64, 64))
+    # # This should update the texture with a surface filled with red
+    # # Instead of creating texture every frame, use this will be less expensive
+    # surf.fill((255,0,0))
+    # tex.update(surf)
```

Comment:
Also, it would likely be a lot faster to create a separate texture for each surface before the loop starts than to update a single one.

Reply:
Is every texture need to be locked before SDL_UpdateTexture, or only those with `streaming=True`?

Also yes, create separate texture for each surface will be faster, but the example is to show you can update a texture with surface.

Reply:
Locking would be used instead of `SDL_UpdateTexture()` for streaming textures. The latter might not update the pixel array properly in case we ever add direct pixel access to `Texture`, according to the wiki. But it can be done later?